### PR TITLE
Feature/volume settings

### PR DESCRIPTION
**Pull request by ondrejmyska**
Draft of proposal with audio volume settings. There is a few conceptual changes I would like to discuss when the feature is interesting for you.

1. volume level is stored in root level of pokete.json file. Originally, I wanted to store it to the settings, but there are only boolean values, not e.g. inteeger values.
2. Menu - audio setting are flat - placed on the same level. Nice to have feature would be have posibility to group related settings: 
  Audio settings
   - Audio: on
   - Audio volume: 100
obrazek
3. Maybe better validation of interval inteeger values. Current implementation is best efford, maybe sufficicent.

Changes are without binary files. I cannot compile all supported architecture. Cross compiling is not working for me. I've tested it with windows rebuilded library.

What do you think @lxgr-linux  about it?

### CODE REVIEW

**Comment by lxgr-linux** [comment]
Ohh, that's very nice.
Actually that's a feature @MaFeLP and I have been talking about since sound was added to the game.

> volume level is stored in root level of pokete.json file. Originally, I wanted to store it to the settings, but there are only boolean values, not e.g. inteeger values.

The settings support all kind of data types, even the `VisSetting`s support setting all values, not just integers. But until now, at least on the master branch, there was no need for non boolean settings, but those are supported by the settingsystem.

> Menu - audio setting are flat - placed on the same level. Nice to have feature would be have posibility to group related settings:
Audio settings

That's very intuitive.

> Maybe better validation of interval inteeger values. Current implementation is best efford, maybe sufficicent.

I also thought about some kind of a nice looking volume slider, able to be manipulated with `a` and `d`.

**Comment by lxgr-linux** [comment]
I will now myself look into integrating with the settings system.

**Comment by lxgr-linux** [comment]
I now involved the settingsystem and made the audio settings reactive, so they apply immediately after change.

**Comment by lxgr-linux** [comment]
> I also thought about some kind of a nice looking volume slider, able to be manipulated with a and d.

I would suggest having a slider like this.

image

**Comment by ondrejmyska** [comment]
> I now involved the settingsystem and made the audio settings reactive, so they apply immediately after change.

Perfect, I forgot to do that

**Comment by ondrejmyska** [comment]
> I now involved the settingsystem and made the audio settings reactive, so they apply immediately after change.

Thanks!

**Comment by ondrejmyska** [comment]
> 



> > I also thought about some kind of a nice looking volume slider, able to be manipulated with a and d.
> 
> I would suggest having a slider like this.
> 
> image

Nice suggestion, I like this idea. Especially, colorful levels are intiutive from UX point of view. Will you add slider in separate PR?

**Comment by lxgr-linux** [comment]
> > 
> 
> > > I also thought about some kind of a nice looking volume slider, able to be manipulated with a and d.
> > 
> > 
> > I would suggest having a slider like this.
> > image
> 
> Nice suggestion, I like this idea. Especially, colorful levels are intiutive from UX point of view. Will you add slider in separate PR?

Yes, I can do that, is this now open for review?

**Comment by ondrejmyska** [comment]
> 



> > > 
> > 
> > 
> > > > I also thought about some kind of a nice looking volume slider, able to be manipulated with a and d.
> > > 
> > > 
> > > I would suggest having a slider like this.
> > > image
> > 
> > 
> > Nice suggestion, I like this idea. Especially, colorful levels are intiutive from UX point of view. Will you add slider in separate PR?
> 
> Yes, I can do that, is this now open for review?

Almost, I think recompiled macOS playsound library libplaysound.osx.so is still missing.

**Comment by lxgr-linux** [comment]
> > 
> 
> > > > 
> > > 
> > > 
> > > > > I also thought about some kind of a nice looking volume slider, able to be manipulated with a and d.
> > > > 
> > > > 
> > > > I would suggest having a slider like this.
> > > > image
> > > 
> > > 
> > > Nice suggestion, I like this idea. Especially, colorful levels are intiutive from UX point of view. Will you add slider in separate PR?
> > 
> > 
> > Yes, I can do that, is this now open for review?
> 
> Almost, I think recompiled macOS playsound library libplaysound.osx.so is still missing.

Since I don't own a mac either and can't crosscompile for OSX I think we have to rely on @MaFeLP

File: playsound/__init__.py [hunk 1]
```diff
@@ -4,13 +4,12 @@
 from pathlib import Path
 
 
-def playsound(file):
+def playsound(file, volume):
     """Plays a mp3 file
     ARGS:
         file: path to file"""
```
Comment (lxgr-linux) [inline playsound/__init__.py:10]:
```suggestion
        file: path to file
        volume: the sound volume as an int 0-100"""
```

File: pokete_classes/audio.py [hunk 2]
```diff
@@ -8,13 +8,13 @@
 MUSIC_PATH = Path(__file__).parents[1] / 'assets' / 'music'
 
 
-def audio_fn(song, play_audio):
+def audio_fn(song, play_audio, volume):
     """plays a song in loop"""
```
Comment (lxgr-linux) [inline pokete_classes/audio.py:12]:
```suggestion
    """plays a song in loop
    ARGS:
        song: path to file
        play_audio: bool whether or not audio actually should be played
        volume: the sound volume as an int 0-100"""
```

**Review by lxgr-linux — CHANGES_REQUESTED** [review]
Just some doc strings.

**Comment by ondrejmyska** [comment]
> Just some doc strings.

Thank's, I've just applied you suggestions directly from Github web interface. Interesting tool :)

**Comment by lxgr-linux** [comment]
> > Just some doc strings.
> 
> Thank's, I've just applied you suggestions directly from Github web interface. Interesting tool :)

Oh yes, it's quite nice, and also automatically makes the person suggesting the changes Co-author.

**Review by lxgr-linux — APPROVED** [review]
LGTM

**Review by MaFeLP — CHANGES_REQUESTED** [review]
LGTM

**Review by MaFeLP — APPROVED** [review]
*LGTM

**Pull request merged**